arrows/vxl/vidl_ffmpeg_video_input.cxx

```diff
@@ -362,6 +426,87 @@ class vidl_ffmpeg_video_input::priv
     return retval;
   } // klv_time
 
+// ------------------------------------------------------------------
+  void push_metadata_to_map(vital::timestamp::frame_t fn)
+  {
+    if (fn >= c_start_at_frame &&
+        (c_stop_after_frame == 0 || fn < c_stop_after_frame) &&
+        c_use_metadata)
+    {
+      metadata_collection.clear();
+      auto curr_md = d_video_stream.current_metadata();
+      if (process_metadata( curr_md ) )
+      {
+        std::pair<vital::timestamp::frame_t, vital::metadata_vector>
+          el(fn, metadata_collection);
+        d_metadata_map.insert( el );
+      }
+    }
+  }
+
+// ------------------------------------------------------------------
+  void process_loop_dependencies()
+  {
+    // is stream open?
+    if ( ! d_video_stream.is_open() )
+    {
+      throw vital::file_not_read_exception( video_path, "Video not open" );
+    }
+
+    if ( !d_have_loop_vars )
+    {
+      if ( d_video_stream.is_seekable() )
+      {
+        std::lock_guard< std::mutex > lock( s_open_mutex );
+
+        if (d_frame_advanced)
+        {
+          d_num_frames = d_frame_number;
+        }
+        else
+        {
+          d_num_frames = 1;
+        }
+
+        // Add metadata for current frame
+        push_metadata_to_map(d_num_frames);
+
+        // Advance video stream to end
+        unsigned int frames_left_to_skip = c_frame_skip - 1;
+        while ( d_video_stream.advance())
+        {
+          d_num_frames++;
+          if (frames_left_to_skip == 0)
+          {
+            push_metadata_to_map(d_num_frames);
+            frames_left_to_skip = c_frame_skip;
+          }
+          frames_left_to_skip--;
+        }
+
+        metadata_collection.clear();
+
+        // Close and reopen to reset
+        d_video_stream.open( video_path );
+
+        // Advance back to original frame number
+        frames_left_to_skip = 0;
```

Comment:
@chetnieter See my other comment on the failing test for context.  I think this might be related to the failing test.  It starts at 1 above, but then goes back to the beginning at `i=0`.  I think this is where the `0` frame metadata is coming from, but I don't understand this code enough to fix it.  Is it just a matter of  changing the loop below to start with `i=1`?